library/std/src/io/impls.rs

```diff
@@ -368,7 +368,10 @@ impl<A: Allocator> Write for Vec<u8, A> {
 
     #[inline]
     fn write_vectored(&mut self, bufs: &[IoSlice<'_>]) -> io::Result<usize> {
-        let len = bufs.iter().map(|b| b.len()).sum();
+        let len = bufs
+            .iter()
+            .try_fold(0usize, |len, b| len.checked_add(b.len()))
+            .ok_or(ErrorKind::OutOfMemory)?;
```

Comment:
this is not a Out of memory, I would prefer to use Other for now.

Reply:
I think OutOfMemory is fine here. We use that error to indicate the result won't fit in available memory. If the number of bytes doesn't fit in a `usize`, it definitely won't fit in memory.

Reply:
oh my bad I through it was isize

Reply:
On a second though why not return an overflow error so ?